lib/eventFactory.js

```diff
@@ -505,13 +545,42 @@ async function createStages({ pipeline, groupEventId }) {
         return Promise.resolve();
     }
 
-    const newStages = await convertStages({ pipeline, stages, groupEventId });
+    // list stage names from this pipeline that already exist
+    const existingStages = await stageFactory.list({ params: { pipelineId, name: stageNames } });
+    const existingStageNames = existingStages.map(stage => stage.name);
+    const convertedStages = await convertStages({ pipeline, stages, eventId, jobs, existingStages });
```

Comment:
`existingStages` is not being used in the function `convertStages()`

Reply:
removed